tests/unit/morphing/test_concrete_provider.py

```diff
@@ -129,6 +157,77 @@ def test_datetime_format_provider(strict_coercion, debug_trail):
     assert dumper(datetime(year=3045, month=2, day=13)) == "3045-02-13"
 
 
+@pytest.mark.parametrize(
+    "tz",
+    [
+        None,
+        timezone(timedelta(hours=3)),
+    ],
+)
+def test_datetime_timestamp_provider(strict_coercion, debug_trail, tz: timezone):
+    retort = Retort(
+        strict_coercion=strict_coercion,
+        debug_trail=debug_trail,
+        recipe=[
+            DatetimeTimestampProvider(tz=tz),
+        ],
+    )
+
+    loader = retort.get_loader(datetime)
+
+    dt = datetime(2011, 11, 4, 6, 38, tzinfo=tz)
+    ts = dt.timestamp()
+
+    assert loader(ts) == dt
+
+    overflow_ts = float("inf")
+    nan = float("nan")
+
+    raises_exc(
+        ValueLoadError("Timestamp is out of the range of supported values", overflow_ts),
+        lambda: loader(overflow_ts),
+    )
+
+    raises_exc(
+        ValueLoadError("Unexpected value", nan),
+        lambda: loader(nan),
+    )
```

Comment:
When you create different values ​​at the beginning, and then test them in a row, you get the impression that these values ​​or tests are somehow logically connected. However, these are completely different errors